From f7fb318a09a0d0b6275a788994d258f4a40d6605 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Wed, 11 Dec 2024 13:46:00 +0100
Subject: [PATCH] Debug fixing stuff

---
 src/SWIG/_ssl.i | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/SWIG/_ssl.i b/src/SWIG/_ssl.i
index 305a556..2b1c4c5 100644
--- a/src/SWIG/_ssl.i
+++ b/src/SWIG/_ssl.i
@@ -476,9 +476,14 @@ static void ssl_handle_error(int ssl_err, int ret) {
             break;
         case SSL_ERROR_SYSCALL:
             err = ERR_get_error();
-            if (err)
-                PyErr_SetString(_ssl_err, ERR_reason_error_string(err));
-            else if (ret == 0)
+            if (err) {
+                char* err_str = ERR_reason_error_string(err);
+                if (!err_str) {
+                    printf("CBURR ERR_get_error() returned %d\n", err);
+                }
+                err_str = err_str ? err_str : "no SSL error string found";
+                PyErr_SetString(_ssl_err, err_str);
+            } else if (ret == 0)
                 PyErr_SetString(_ssl_err, "unexpected eof");
             else if (ret == -1)
                 PyErr_SetFromErrno(_ssl_err);
-- 
2.47.0

