diff --git a/setup.py b/setup.py
index 38b5117..c741399 100644
--- a/setup.py
+++ b/setup.py
@@ -128,7 +128,10 @@ class _M2CryptoBuildExt(build_ext.build_ext):
         # type: (None) -> None
         """Append custom openssl include file and library linking options."""
         build_ext.build_ext.finalize_options(self)
-        self.openssl_default = None
+        if sys.platform == 'win32':
+            self.openssl_default = os.getenv('LIBRARY_PREFIX')
+        else:
+            self.openssl_default = os.getenv('PREFIX')
         self.set_undefined_options('build', ('openssl', 'openssl'))
         if self.openssl is None:
             self.openssl = self.openssl_default
