From 241d7298e0b5f2530c275063db98e05ee0ab00df Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Thu, 15 Dec 2022 09:02:08 +0100
Subject: [PATCH] Remove buggy ifdef statements (m2crypto#298)

---
 src/SWIG/_x509.i | 36 ------------------------------------
 1 file changed, 36 deletions(-)

diff --git a/src/SWIG/_x509.i b/src/SWIG/_x509.i
index b6334df..9246cc2 100644
--- a/src/SWIG/_x509.i
+++ b/src/SWIG/_x509.i
@@ -347,62 +347,26 @@ X509 *d2i_x509(BIO *bio) {
 
 /* See man page of X509_VERIFY_PARAM_set_flags for definition of all these flags */
 
-#ifdef X509_V_FLAG_ALLOW_PROXY_CERTS
 %constant int VERIFY_ALLOW_PROXY_CERTS  = X509_V_FLAG_ALLOW_PROXY_CERTS;
-#endif
-#ifdef X509_V_FLAG_CB_ISSUER_CHECK
 %constant int VERIFY_CB_ISSUER_CHECK  = X509_V_FLAG_CB_ISSUER_CHECK;
-#endif
-#ifdef X509_V_FLAG_CHECK_SS_SIGNATURE
 %constant int VERIFY_CHECK_SS_SIGNATURE  = X509_V_FLAG_CHECK_SS_SIGNATURE;
-#endif
 /* note: X509_V_FLAG_CRL_CHECK is already defined in _ssl.i as VERIFY_CRL_CHECK_LEAF
 However I add it here for consistency */
-#ifdef X509_V_FLAG_CRL_CHECK
 %constant int VERIFY_CRL_CHECK  = X509_V_FLAG_CRL_CHECK;
-#endif
-#ifdef X509_V_FLAG_CRL_CHECK_ALL
 %constant int VERIFY_CRL_CHECK_ALL  = X509_V_FLAG_CRL_CHECK_ALL;
-#endif
-#ifdef X509_V_FLAG_EXPLICIT_POLICY
 %constant int VERIFY_EXPLICIT_POLICY  = X509_V_FLAG_EXPLICIT_POLICY;
-#endif
-#ifdef X509_V_FLAG_EXTENDED_CRL_SUPPORT
 %constant int VERIFY_EXTENDED_CRL_SUPPORT  = X509_V_FLAG_EXTENDED_CRL_SUPPORT;
-#endif
-#ifdef X509_V_FLAG_IGNORE_CRITICAL
 %constant int VERIFY_IGNORE_CRITICAL  = X509_V_FLAG_IGNORE_CRITICAL;
-#endif
-#ifdef X509_V_FLAG_INHIBIT_ANY
 %constant int VERIFY_INHIBIT_ANY  = X509_V_FLAG_INHIBIT_ANY;
-#endif
-#ifdef X509_V_FLAG_INHIBIT_MAP
 %constant int VERIFY_INHIBIT_MAP  = X509_V_FLAG_INHIBIT_MAP;
-#endif
-#ifdef X509_V_FLAG_NO_ALT_CHAINS
 %constant int VERIFY_NO_ALT_CHAINS  = X509_V_FLAG_NO_ALT_CHAINS;
-#endif
-#ifdef X509_V_FLAG_NO_CHECK_TIME
 %constant int VERIFY_NO_CHECK_TIME  = X509_V_FLAG_NO_CHECK_TIME;
-#endif
-#ifdef X509_V_FLAG_NOTIFY_POLICY
 %constant int VERIFY_NOTIFY_POLICY  = X509_V_FLAG_NOTIFY_POLICY;
-#endif
-#ifdef X509_V_FLAG_PARTIAL_CHAIN
 %constant int VERIFY_PARTIAL_CHAIN  = X509_V_FLAG_PARTIAL_CHAIN;
-#endif
-#ifdef X509_V_FLAG_POLICY_CHECK
 %constant int VERIFY_POLICY_CHECK  = X509_V_FLAG_POLICY_CHECK;
-#endif
-#ifdef X509_V_FLAG_TRUSTED_FIRST
 %constant int VERIFY_TRUSTED_FIRST  = X509_V_FLAG_TRUSTED_FIRST;
-#endif
-#ifdef X509_V_FLAG_USE_DELTAS
 %constant int VERIFY_USE_DELTAS  = X509_V_FLAG_USE_DELTAS;
-#endif
-#ifdef X509_V_FLAG_X509_STRICT
 %constant int VERIFY_X509_STRICT  = X509_V_FLAG_X509_STRICT;
-#endif
 
 
 
-- 
2.38.1

