diff --git a/setup.py b/setup.py
index 7efd6d4..0327d59 100644
--- a/setup.py
+++ b/setup.py
@@ -50,14 +50,7 @@ def _get_additional_includes():
                                 '*Visual*', 'VC', 'include')
         err = glob.glob(globmask)
     else:
-        pid = subprocess.Popen(['cpp', '-Wp,-v', '-'],
-                               stdin=open(os.devnull, 'r'),
-                               stdout=subprocess.PIPE,
-                               stderr=subprocess.PIPE)
-        _, err = pid.communicate()
-        err = [line.lstrip() for line in err.decode('utf8').split('\n')
-               if line and line.startswith(' /')]
-
+        err = []
     log.debug('additional includes:\n%s', err)
     return err